File: Data Structures & Algorithms/binary-search/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {

         let mem = new Map()

         for(let i = 0; i<nums.length;i++){
            mem.set(nums[i],i)
         }



        while(nums.length >0){

            let halfIndex = Math.floor(nums.length/2)
            console.log('\n',nums[halfIndex])

            if(target === nums[halfIndex]){return mem.get(nums[halfIndex])}

            if(target > nums[halfIndex]){

                nums = nums.slice(halfIndex+1, nums.length)
                console.log('greater than half so: ', nums)
            }else if(target < nums[halfIndex] ){

                nums = nums.slice(0, halfIndex)
                console.log('less than half so: ', nums)

            }else if(target === nums[halfIndex]){return mem.get(nums[halfIndex])}

        } 

        return -1
      
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.java


class Solution {
    public boolean hasDuplicate(int[] nums) {

        Set<Integer> mem = new HashSet<>();

        for(int i = 0; i<nums.length;i++){
            if(mem.contains(nums[i])){
                return true;
            }else{
                mem.add(nums[i]);
            }
        }
        return false;
    }
}